beprof/curve.py

```diff
@@ -66,19 +67,98 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+        '''
+        Creating new Curve object in memory with domain passed as a parameter.
+        New domain must include in the orignal domain.
+        Copies values from orginal curve and uses interpolation to calculate
+        values for new points in domain.
+
+        Calculate y - values of example curve with changed domain:
+        >>> print(Curve([[0,0], [5, 5], [10, 0]]).change_domain([1, 2, 8, 9]).y)
+        [ 1.  2.  2.  1.]
+
+        :param domain: set of points representing new domain. Might be a list or np.array
+        :return: new Curve object with domain set by 'domain' parameter
+        '''
+
+        # check if new domain includes in the orginal domain
+        if np.max(domain) > np.max(self.x):
+            # separate issue created to provide logging package
+            print('Error1')
+            return self
+        if np.min(domain) < np.min(self.x):
+            print('Error2')
+            return self
+        y = np.interp(domain, self.x, self.y)
+        obj = Curve(np.stack((domain, y), axis=1))
+        return obj
+
+    def rebinned(self, step=0.1, fixp=0):
+        '''
+        Provides effective way to compute new domain basing on step and fixp parameters.
+        Then using change_domain() method to create new object with calculated domain and returns it.
+
+        fixp doesn't have to be inside orginal domain.
+
+        Return domain of a new curve specified by fixp=0 and step=1 and another Curve object:
+        >>> print(Curve([[0,0], [5, 5], [10, 0]]).rebinned(1,0).x)
+        [  0.   1.   2.   3.   4.   5.   6.   7.   8.   9.  10.]
+
+        :param step: step size of new domain
+        :param fixp: fixed point one of the points in new domain
+        :return: new Curve object with domain specified by step and fixp parameters
+        '''
+        a, b = (np.min(self.x), np.max(self.x))
+        count_start = (abs(fixp - a) / step)
+        count_stop = (abs(fixp - b) / step)
```

Comment:
I have never paid so much attention to how does the code look like but never published one before as well. Brackets no needed at all.